### ISSUE TITLE
[Bug] RSS: Japanese video description encoding issues

### ISSUE THREAD

**Issue by CatPlanet** [iv-org/invidious#4256]
**Describe the bug**
Despite RSS channels' output being in UTF8 japanese video descriptions are total gibberish

**Steps to Reproduce**
1. Go to https://yewtu.be/feed/channel/UCcp9uRaBwInxl_SZqGRksDA
2. Cry

**Comment by unixfox** [comment]
Can you replicate the issue locally with a local instance of invidious?

**Comment by syeopite** [comment]
Seems like a behavior specific to `yewtu.be`. I can't replicate this locally
![replicate](https://github.com/iv-org/invidious/assets/70992037/142114e7-82c5-40ec-8437-42280c1de2a1)

**Comment by RadoslavL** [comment]
> Seems like a behavior specific to `yewtu.be`. I can't replicate this locally ![replicate](https://user-images.githubusercontent.com/70992037/282316093-142114e7-82c5-40ec-8437-42280c1de2a1.png)

I can also replicate it in `invidious.fdn.fr` and `inv.tux.pizza`.

![corrupted-characters](https://github.com/iv-org/invidious/assets/63926126/a9a08fa4-a67c-4090-805b-80d13ef2d76a)

**Comment by absidue** [comment]
I wonder if there is something between Invidious and the user on those major instances that is messing up the encoding, maybe a reverse proxy or something that is incorrectly configured?

**Comment by unixfox** [comment]
@syeopite did you try if you can replicate the issue in docker too?

**Comment by CatPlanet** [comment]
Chinese channels are affected too https://yewtu.be/feed/channel/UCoC47do520os_4DBMEFGg4A
Taiwan https://yewtu.be/feed/channel/UC8ROUUjHzEQm-ndb69CX8Ww
Korean https://yewtu.be/feed/channel/UC5BMQOsAB8hKUyHu9KI6yig
Vietnam https://yewtu.be/feed/channel/UCOYdjTcOmle5oItjpEK0VpQ
Arabic noodles https://yewtu.be/feed/channel/UCfiwzLy-8yKzIbsmZTzxDgw

Titles, names are formatted correctly, same with actual video pages (not RSS ones). It's only the RSS description issue.

**Comment by syeopite** [comment]
Invidious under docker is still able to render the RSS feeds correctly for me. But I am able to replicate this on public instances like `onion.tube` though. 

Is there an instance where the feed is rendered correctly?

**Comment by RadoslavL** [comment]
> Does it work correctly on the test instance? afaik that is a default setup without any of the custom/non-standard stuff that most public instances do.

`/feed/channel/UCoC47do520os_4DBMEFGg4A` doesn't work properly on the test instance.
`/feed/channel/UC8ROUUjHzEQm-ndb69CX8Ww` doesn't work as well.

**Comment by absidue** [comment]
Maybe it's caused by the reverse proxy then?

**Comment by sefidel** [comment]
Can replicate this issue (NixOS 24.05), it's also happening on `/api/v1/comments`.

I've observed the same behaviour without reverse proxy (`localhost` also suffers from this problem).
Perhaps it's something with the build system?

Using the docker image works fine.

**Comment by sefidel** [comment]
This is most likely caused by an outdated version of Crystal.

See https://github.com/crystal-lang/crystal/issues/13703 for more details. ([Relevant Comment](https://github.com/crystal-lang/crystal/issues/13703#issuecomment-1649705973))

**Comment by sefidel** [comment]
The web interface doesn’t suffer from this problem since it uses the ‘contentHtml’ field, which doesn’t go through the XML parser.In other words, I’m 99% certain that this is caused by outdated version of crystal. On 27 Feb 2024, at 17:25, kubo6472 ***@***.***> wrote:﻿
Possibly related, I found that the instance is okay, but the API is apparently the issue: lamarios/clipious#491

**Comment by iGerman00** [comment]
I'm able to reproduce this issue locally, behind Caddy as the reverse proxy, and through Tailscale - I'm pretty sure it's coming out of Invidious that way. I use the Docker image`quay.io/invidious/invidious:latest-arm64` as the way to run Invidious. Is there any specific fix? What should the instance admin fix exactly?

**Comment by sefidel** [comment]
> @iGerman00 Is there any specific fix? What should the instance admin fix exactly?

Compiling invidious with Crystal `>=1.10.0` should solve the issue.
If you want to keep using that container image, I'd suggest opening a new issue/PR to bump the crystal version on the Dockerfile. (looks like the image uses `1.8.2`)

**Comment by syeopite** [comment]
The crystal version for the docker images will get bumped once https://github.com/iv-org/invidious/pull/4654 is merged in the next batch.

**Comment by syeopite** [comment]
#4654 has been merged and as such this should be fixed.
